lucene/core/src/java/org/apache/lucene/search/SortField.java

```diff
@@ -172,8 +180,9 @@ public SortField readSortField(DataInput in) throws IOException {
     }
 
     @Override
-    public void writeSortField(SortField sf, DataOutput out) throws IOException {
-      sf.serialize(out);
+    public void writeSortField(SortOrder sf, DataOutput out) throws IOException {
+      assert sf instanceof SortField;
+      ((SortField)sf).serialize(out);
```

Comment:
And then these spooky `assert` / casts wouldn't be needed -- `javac` would do it for us?